.github/workflows/lint.yml

```diff
@@ -0,0 +1,17 @@
+name: Lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
```

Comment:
Would propose that lint checks should be run for all pushes, if relevant file types change

Reply:
Okay I'll add to run only if changes in `**.go`

Reply:
I removed other branches than master because that was causing the same check to run twice once for your push on the branch and one for the PR that you created with that branch. So for me it made sense to run it once the PR is merged (push to master)
I think its good to run in case when multiple PRs are been worked on the same time and maybe single one doesn't bring the issue but altogether might

Reply:
@BrendanGalloway Any other file types linted with the checker?

Reply:
> I removed other branches than master because that was causing the same check to run twice once for your push on the branch and one for the PR that you created with that branch. So for me it made sense to run it once the PR is merged (push to master)
I think its good to run in case when multiple PRs are been worked on the same time and maybe single one doesn't bring the issue but altogether might

I've always just found the fast feedback useful for when I'm working on a branch. Given the low cost of linting, seems fine to me to run it again on PR. What does everyone else think? @BrendanGalloway @moshloop 

Would running on push for changes without branch scoping differentiate between normal push and PR?

Reply:
I would run it on PR but not on push if we only want to run once?  Everything should be entering master via PR, so it will get checked.

and golang-lint only lints go files as far as I know

Reply:
From a developer feedback perspective, would it not be valuable to run before a developer gets as far as making a PR?

Reply:
From a developer perspective I would say the best would be run it on his local machine instead on a GH action for faster feedback

Reply:
@BrendanGalloway so we remove it from pushes to any branch and just let it be on PR?

Reply:
> From a developer perspective I would say the best would be run it on his local machine instead on a GH action for faster feedback

There is nothing stopping this but it requires manual action. CI is automated. The more automated things check my work the fewer I have to think about at any time.

I still think we should always lint on CI on changes at least.

Reply:
For simple stuff like linting it's probably ok for it to run on push, if it's rerun on PR it's not a big resource consumption.  Was thinking more about the more resource intensive test setups we have